src/main/kotlin/no/nav/familie/ef/sak/iverksett/oppgaveterminbarn/ForberedOppgaverTerminbarnService.kt

```diff
@@ -0,0 +1,88 @@
+package no.nav.familie.ef.sak.iverksett.oppgaveterminbarn
+
+import no.nav.familie.ef.sak.fagsak.FagsakService
+import no.nav.familie.ef.sak.iverksett.IverksettClient
+import no.nav.familie.ef.sak.opplysninger.personopplysninger.PersonService
+import no.nav.familie.ef.sak.opplysninger.personopplysninger.domene.BarnMedIdent
+import no.nav.familie.ef.sak.opplysninger.personopplysninger.mapper.GrunnlagsdataMapper
+import no.nav.familie.ef.sak.opplysninger.personopplysninger.pdl.gjeldende
+import no.nav.familie.kontrakter.ef.iverksett.OppgaveForBarn
+import no.nav.familie.kontrakter.ef.iverksett.OppgaverForBarnDto
+import no.nav.familie.kontrakter.ef.søknad.Fødselsnummer
+import no.nav.familie.kontrakter.felles.ef.StønadType
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDate
+import java.util.UUID
+
+@Service
+class ForberedOppgaverTerminbarnService(private val personService: PersonService,
+                                        private val fagsakService: FagsakService,
+                                        private val terminbarnRepository: TerminbarnRepository,
+                                        private val iverksettClient: IverksettClient) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    @Transactional
```

Comment:
Hvorfor `Transactional` her? Burde denne gjøres i en ny transaksjon?

Reply:
Tenkte å ha transactional siden vi lagrer ned TerminbarnOppgave før vi sender til iverksett (og veldig få oppgaver i antall) ? Men ja, vi må kanskje ha en ny transaksjon med REQUIRES_NEW i så fall ?

Reply:
Denne kalles alltid fra en task, som er i transactional.
Det er fint å markere metoder som transactional for å gjøre det tydelig at det skjer i en transaction, men når man gjører flere http-kall og databasoppdateringer må man vurdere hvordan feilhåndteringen skal skje.

Eks så hadde man kunnet flyttet ut `sendOppgaverTilIverksett(oppgaver)` utenfor `forEach` og kun sende oppgaver en gang, og då er greit med å kun ha `Transactional`.
Anbefaler nog også å endre til å gjøre det slik